### PR TITLE
[BB-1073] add travis & tox for CI

### PR DESCRIPTION
**Pull request by dupainaulevain**

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @dupainaulevain! I've created [OSPR-3242](https://openedx.atlassian.net/browse/OSPR-3242) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by dupainaulevain** [comment]
@xitij2000 would it be possible to configure travis on this repository? Or should this be tested in another repository instead?

**Comment by dupainaulevain** [comment]
@giovannicimolin ^

**Comment by xitij2000** [comment]
@natabene Would you be able to enable running TravisCI tests on this repository? We'd like to enable automated tests for a few other repos on edx-solutions as well, but this is a start.

**Comment by dupainaulevain** [comment]
@natabene I hope you enjoyed the conference last week ! I assume you're busy going through the backlog of notifications. This is just a gentle ping to let you know your help would be very welcome here :-)

**Comment by natabene** [comment]
@jmbowman I do not know how to run travis CI, am I even able to? Can you run it for me or let me know how to do it?

**Comment by natabene** [comment]
> @natabene I hope you enjoyed the conference last week ! I assume you're busy going through the backlog of notifications. This is just a gentle ping to let you know your help would be very welcome here :-)

Sorry, I never ran it, so no idea how to do it. i Asked Jeremy. 
Thanks, conference was great!

**Comment by natabene** [comment]
@dupainaulevain We made changes to the webhooks. Can you do a new push of the PR? We'll see if it works now.

**Comment by giovannicimolin** [comment]
@natabene I asked for permissions to install Travis CI on this repo. I think that'll make it work, can you approve it?

**Comment by natabene** [comment]
> @natabene I asked for permissions to install Travis CI on this repo. I think that'll make it work, can you approve it?

I think I just approved, please verify

**Comment by giovannicimolin** [comment]
@natabene Thanks for that! :rocket: 
@dupainaulevain You need to open a pull request from a branch on `edx-solutions` for the CI checks to run. I pushed your branch to the repo and triggered a build.

**Comment by dupainaulevain** [comment]
@giovannicimolin @natabene thanks ! 

@giovannicimolin I unfortunately do not have permission to push on this repository:

```
128 git … push -v origin dachary/BB-1073-travis\:refs/heads/dachary/BB-1073-travis
Pushing to git@github.com:edx-solutions/xblock-image-explorer.git
ERROR: Permission to edx-solutions/xblock-image-explorer.git denied to dupainaulevain.
```

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+dist: xenial
```
Comment (giovannicimolin) [inline .travis.yml]: Can you use a python container to start builds? ([Running python container with geckodriver](http://www.columbia.edu/~njn2118/journal/2016/10/28.html)). This file will be way simpler that way.

File: README.md [hunk 2]
```diff
@@ -35,26 +35,13 @@ advanced settings.
...
-
-To run the unit tests, run:
-
-```bash
-python run_tests.py tests/unit
-```
+2. mkvirtualenv xblock-image-explorer
 
-To run the integration tests, you'll need [`geckodriver`](https://github.com/mozilla/geckodriver) and `xvfb` installed.
+3. pip install tox
```
Comment (giovannicimolin) [inline README.md]: Same thing here, you don't need to force the developer to install geckodriver and run the integration tests locally.

File: README.md [hunk 3]
```diff
@@ -35,26 +35,13 @@ advanced settings.
 Testing
 -------
 
-Make sure you have a python virtual environment with `xblock-sdk` version `0.1.5` or higher:
+1. Mimic the installation instructions for firefox & geckodriver from .travis.yml
```
Comment (giovannicimolin) [inline README.md]: If you are going to edit testing instructions, please to so in a explicit way in the documentation. There's no need to start from xenial and install all the requirements that Travis needs in order to run the tests.

**Pull request closed without merging**